packages/contentstack-import/src/import/modules/marketplace-apps.ts

```diff
@@ -373,15 +422,16 @@ export default class ImportMarketplaceApps {
 
     if (!currentStackApp) {
       // NOTE install new app
-      const mappedUid = this.appUidMapping[app?.manifest?.uid] as string;
-      const installation = await installApp(this.sdkClient, this.importConfig, app?.manifest?.uid, mappedUid);
+      const installation = await this.installApp(
+        this.importConfig,
+        // NOTE if it's private app it should get uid from mapper else will use manifest uid
+        this.appUidMapping[app.manifest.uid] || app.manifest.uid,
+      );
 
       if (installation.installation_uid) {
-        const appManifestName = app?.manifest?.name;
-        const appName = this.appNameMapping[appManifestName] ? this.appNameMapping[appManifestName] : appManifestName;
-
+        const appName = this.appNameMapping[app.manifest.name] ?? app.manifest.name;
         log(this.importConfig, `${appName} app installed successfully.!`, 'success');
-        await makeRedirectUrlCall(installation, appManifestName, this.importConfig);
+        await makeRedirectUrlCall(installation, app.manifest.name, this.importConfig);
```

Comment:
@antonyagustine , The app name has been assigned to the appName variable at Line 432. Thus, rather of using app.manifest.name in Line 434, should we use this?